rtrlib/aspa/aspa_array/aspa_array.c

```diff
@@ -228,3 +222,35 @@ struct aspa_record *aspa_array_search(struct aspa_array *array, uint32_t custome
 	// element not found
 	return NULL;
 }
+
+
+enum aspa_status aspa_array_reserve(struct aspa_array *array, size_t size) {
+	// the given array is null
+	if (array == NULL) {
+		return ASPA_ERROR;
+	}
+
+	// we already have enough space allocated
+	if (array->capacity >= size) {
+		return ASPA_SUCCESS;
+	}
+
+	struct aspa_record* data = malloc(sizeof(struct aspa_record) * size);
+
+	// malloc failed
+	if (data == NULL) {
+		return ASPA_ERROR;
+	}
+
+	// coping the data from the old array into the new one
+	if (memcpy(data, array->data, array->size * sizeof(struct aspa_record)) == NULL) {
+		return ASPA_ERROR;
+	}
```

Comment:
Can't we use realloc/reallocarray here instead of malloc+ memcpy?

Reply:
done